sherpa/plot/backends.py

```diff
@@ -17,19 +17,88 @@
 #  with this program; if not, write to the Free Software Foundation, Inc.,
 #  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 #
+import functools
+from inspect import signature
 import logging
 
-from sherpa.utils import get_keyword_defaults
 from sherpa.utils import formatting
 
 
-__all__ = ('BasicBackend')
+__all__ = ('BasicBackend',
+           'backend_indep_kwargs',
+           )
 
 
 lgr = logging.getLogger(__name__)
 warning = lgr.warning
 
 
+# Should None be an options for all of them?
+backend_indep_kwargs = {
+    'color': 'rgbkwcymk',
+    'ecolor': 'rgbkwcymk',
+    'markerfacecolor': 'rgbkwcymk',
+    'linestyle': ['noline', 'solid', 'dot', 'dash', 'dotdash', '-', ':', '--',
+                  '-.', '', None],
+    'marker': ['', "None", ".", "o", "+", "s"]
+}
+
+
+def translate_args(func):
+    @functools.wraps(func)
+    def inner(self, *args, **kwargs):
+        for kw, val in kwargs.items():
+            if kw in self.translate_args:
+                transl = self.translate_kwargs[kw]
+
+                if callable(transl):
+                    # It's a function
+                    kwargs[kw] = transl(val)
+                else:
+                    # It should be a dict
+                    # Let's check if val is one of those that need to
+                    # be translated
+                    if val in transl:
+                        kwargs[kw] = transl[val]
+        func(self, *args, **kwargs)
+
+    return inner
+
+# In principle, still need test for get_keyword_default
+# However, I expect to remove that in step 2 of the plotting plan,
+# so I don't want to spend too much time on that right now.
+
+
+def get_keyword_defaults(func, ignore_args=['title', 'xlabel', 'ylabel',
```

Comment:
I was going to suggest removing `sherpa.utils.get_keyword_defaults` but then I checked and it's used in other places than just plotting. It would be helpful to have a note in the docs here about how this is different from the utils version. I assume it's primarily the `ignore_args` logic.